c/src/ml-api-common.c

```diff
@@ -1647,3 +1647,202 @@ ml_option_get (ml_option_h option, const char *key, void **value)
 
   return _ml_info_get_value ((ml_info_s *) option, key, value);
 }
+
+/**
+ * @brief Creates an ml_information instance and returns the handle.
+ */
+int
+_ml_information_create (ml_information_h * info)
+{
+  ml_info_s *_info = NULL;
+
+  check_feature_state (ML_FEATURE);
+
+  if (!info) {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "The parameter, 'info' is NULL. It should be a valid ml_information_h");
+  }
+
+  _info = _ml_info_create (ML_INFO_MODEL);
+  if (!_info)
+    _ml_error_report_return (ML_ERROR_OUT_OF_MEMORY,
+        "Failed to allocate memory for the info handle. Out of memory?");
+
+  *info = (ml_information_h *) _info;
+  return ML_ERROR_NONE;
+}
+
+/**
+ * @brief Set key-value pair in given information handle.
+ * @note If duplicated key is given, the value is updated with the new one.
+ */
+int
+_ml_information_set (ml_information_h information, const char *key, void *value,
+    ml_data_destroy_cb destroy)
```

Comment:
Is ml_data_destroy_cb nullable?

Reply:
Yes, it is nullable, if user wants handing the value by himself.